integration_tests/test_dbapi.py

```diff
@@ -116,7 +116,42 @@ def test_select_failed_query(presto_connection):
         cur.execute("select * from catalog.schema.do_not_exist")
         cur.fetchall()
 
+def test_select_query_result_iteration_statement_params(presto_connection):
+    cur = presto_connection.cursor()
+    cur.execute(
+        """
+        select * from (
+            values
+            (1, 'one', 'a'),
+            (2, 'two', 'b'),
+            (3, 'three', 'c'),
+            (4, 'four', 'd'),
+            (5, 'five', 'e')
+        ) x (id, name, letter)
+        where id >= ?
+        """,
+        params=(3,)  # expecting all the rows with id >= 3
```

Comment:
can you add a test case covering dates, bytes and other types mentioned in `def _format_prepared_param`

Reply:
done